### PR TITLE
[ Magiclysm ] Fortifying the forge

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: Mods "Forge of Wonders Gygaxian Update"

#### Purpose of change
Close up a number of exploits for the Forge of Wonders.  I've listed this as none so it won't be obvious in the change log.  TPK gygaxian dungeon here we come! Contributes to #43018

#### Describe the solution
Adds randomized traps to the vault area.  Changes some easily abused artifacts if I can think of ways to make them less abuseable.

#### Describe alternatives you've considered

none

#### Testing
It works! TPK inbound!

### CODE REVIEW

**Comment by waveyl** [comment]
still a bug.
spider box

**Comment by Maleclypse** [comment]
> still a bug.
> spider box
> as well as enchanted tailor's kit.

Not a bug I can address in this PR since that's got something to do with C++ valuing items with zero charges as worthless.

**Comment by KorGgenT** [comment]
make sure you leave a comment when you make this not wip again - i don't receive an email update when pr titles change

**Comment by keampe** [comment]
So, I tried my hand at fixing some of this. For the Spider Box and the Enchanted Tailors kit I changed the "itemgroups.json" lines 318&319 to add the "charges-min" and "charges-max". This gave the items a charge to start with, allowing the price not to be 0.
Also, these items have a "price" but not a ""price_postapoc". This might account for the cheapness of buying them.

   - Shane

**Comment by Maleclypse** [comment]
@curstwist I'm not having any luck with these traps. I'm not sure if it's due to traps not rotating with the maps or what?  Sometimes monsters will appear but that is super rare and once monsters appeared everywhere but only once and I've never been able to repeat that.

**Comment by curstwist** [comment]
> 
> 
> @curstwist I'm not having any luck with these traps. I'm not sure if it's due to traps not rotating with the maps or what? Sometimes monsters will appear but that is super rare and once monsters appeared everywhere but only once and I've never been able to repeat that.

yeah, it's an issue with merged maps not supporting trap placement/spawning beyond the first OMT.
You can break it up and it should work fine.  I'd suggest just breaking out the 2 OMTs that have the vault into their own maps instead of breaking it all up.

I discovered this really painful bug while messing with new lab map thoughts and already reported it.

**Comment by Maleclypse** [comment]
> > @curstwist I'm not having any luck with these traps. I'm not sure if it's due to traps not rotating with the maps or what? Sometimes monsters will appear but that is super rare and once monsters appeared everywhere but only once and I've never been able to repeat that.
> 
> yeah, it's an issue with merged maps not supporting trap placement/spawning beyond the first OMT.
> You can break it up and it should work fine. I'd suggest just breaking out the 2 OMTs that have the vault into their own maps instead of breaking it all up.
> 
> I discovered this really painful bug while messing with new lab map thoughts and already reported it.

So when I break them out into individual OMT's do I have to remove them from the merged map? If so I'll probably have to break them all out for my own sanity.

File: data/mods/Magiclysm/traps.json [hunk 1]
```diff
@@ -11,5 +11,51 @@
...
+    "map_regen": "forge_trap2",
+    "benign": false
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "forge_trap2",
+    "method": "json",
+    "object": {
+      "place_terrain": [ { "ter": "t_strconc_floor", "x": 76, "y": [ 8, 14 ] } ],
+      "place_monsters": [ { "monster": "GROUP_FORGEDEMONS", "x": [ 73, 77 ], "y": [ 7, 14 ], "density": 0.2, "repeat": [ 1, 3 ] } ]
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traps.json:47]:
```suggestion
      "place_monsters": [ { "monster": "GROUP_FORGEDEMONS", "x": [ 73, 77 ], "y": [ 7, 14 ], "density": 0.5, "repeat": [ 1, 3 ] } ]
```

File: data/mods/Magiclysm/traps.json [hunk 2]
```diff
@@ -11,5 +11,51 @@
...
+    }
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "forge_trap1",
+    "method": "json",
+    "object": {
+      "place_monsters": [
+        { "monster": "GROUP_FORGEDEMONS", "x": [ 72, 94 ], "y": [ 7, 14 ], "density": 0.2, "repeat": [ 1, 3 ] },
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traps.json:56]:
```suggestion
        { "monster": "GROUP_FORGEDEMONS", "x": [ 72, 94 ], "y": [ 7, 14 ], "density": 0.5, "repeat": [ 1, 3 ] },
```

File: data/mods/Magiclysm/traps.json [hunk 3]
```diff
@@ -11,5 +11,51 @@
...
+        { "monster": "GROUP_FORGEDEMONS", "x": [ 96, 117 ], "y": [ 7, 14 ], "density": 0.2, "repeat": [ 1, 3 ] }
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traps.json:57]:
```suggestion
        { "monster": "GROUP_FORGEDEMONS", "x": [ 96, 117 ], "y": [ 7, 14 ], "density": 0.5, "repeat": [ 1, 3 ] }
```

**Comment by Maleclypse** [comment]
@KorGgenT  I'm really happy with this and will use these kinds of effects more often now :)  Ready for merge at your discretion.

File: data/mods/Magiclysm/items/enchanted.json [hunk 4]
```diff
@@ -3,13 +3,15 @@
     "id": "belt_str",
     "copy-from": "leather_belt",
     "type": "ARMOR",
+    "price_postapoc": 175000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted.json:6]:
```suggestion
    "price_postapoc": "1750 USD",
```

File: data/mods/Magiclysm/items/enchanted.json [hunk 5]
```diff
@@ -3,13 +3,15 @@
...
     "name": { "str": "belt of strength +2", "str_pl": "belts of strength +2" },
     "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "STRENGTH", "add": 2 } ] } ] }
   },
   {
     "id": "belt_haste",
     "copy-from": "leather_belt",
     "type": "ARMOR",
+    "price_postapoc": 350000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted.json:14]:
```suggestion
    "price_postapoc": "3500 USD",
```

File: data/mods/Magiclysm/items/enchanted.json [hunk 6]
```diff
@@ -42,6 +44,7 @@
     "weight": "390 g",
     "volume": "2 L",
     "price": 9500,
+    "price_postapoc": 120000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted.json:47]:
```suggestion
    "price_postapoc": "1200 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 7]
```diff
@@ -7,6 +7,7 @@
     "weight": "4 g",
     "volume": "1 ml",
     "price": 5000,
+    "price_postapoc": 2000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:10]:
```suggestion
    "price_postapoc": "20 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 8]
```diff
@@ -24,6 +25,7 @@
     "weight": "5 g",
     "volume": "1 ml",
     "price": 5000,
+    "price_postapoc": 2000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:28]:
```suggestion
    "price_postapoc": "20 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 9]
```diff
@@ -41,6 +43,7 @@
     "weight": "9 g",
     "volume": "1 ml",
     "price": 5000,
+    "price_postapoc": 2000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:46]:
```suggestion
    "price_postapoc": "20 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 10]
```diff
@@ -58,6 +61,7 @@
     "weight": "11 g",
     "volume": "1 ml",
     "price": 5000,
+    "price_postapoc": 2000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:64]:
```suggestion
    "price_postapoc": "20 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 11]
```diff
@@ -73,12 +77,14 @@
     "id": "mring_wizardry_minor",
     "name": { "str": "minor ring of wizardry", "str_pl": "minor rings of wizardry" },
     "description": "A thin copper band ring, engraved with a single sealed scroll.  Increases mana capacity slightly.",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:80]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 12]
```diff
@@ -73,12 +77,14 @@
...
     "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "MAX_MANA", "add": 200 } ] } ] }
   },
   {
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_wizardry_lesser",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:87]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 13]
```diff
@@ -88,6 +94,7 @@
     "type": "TOOL_ARMOR",
     "id": "mring_wizardry_moderate",
     "name": { "str": "ring of wizardry", "str_pl": "rings of wizardry" },
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:97]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 14]
```diff
@@ -96,6 +103,7 @@
     "type": "TOOL_ARMOR",
     "id": "mring_wizardry_greater",
     "name": { "str": "greater ring of wizardry", "str_pl": "greater rings of wizardry" },
+    "price_postapoc": 800000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:106]:
```suggestion
    "price_postapoc": "8000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 15]
```diff
@@ -104,6 +112,7 @@
     "type": "TOOL_ARMOR",
     "id": "mring_blades_lesser",
     "name": { "str": "ring of blades", "str_pl": "rings of blades" },
+    "price_postapoc": 150000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:115]:
```suggestion
    "price_postapoc": "1500 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 16]
```diff
@@ -116,13 +125,15 @@
     "type": "TOOL_ARMOR",
     "id": "mring_dodge_bonus_1",
     "name": "eel ring",
+    "price_postapoc": 150000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:128]:
```suggestion
    "price_postapoc": "1500 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 17]
```diff
@@ -116,13 +125,15 @@
...
     "description": "A thin silver band ring, depicting an eel coiled on itself.  Allows you to dodge an extra attack per turn.",
     "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "BONUS_DODGE", "add": 1 } ] } ] }
   },
   {
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_dodge_bonus_2",
+    "price_postapoc": 300000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:136]:
```suggestion
    "price_postapoc": "3000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 18]
```diff
@@ -131,6 +142,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_strength_1",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:145]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 19]
```diff
@@ -139,6 +151,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_strength_2",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:154]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 20]
```diff
@@ -147,6 +160,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_strength_3",
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:163]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 21]
```diff
@@ -155,6 +169,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_strength_4",
+    "price_postapoc": 600000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:172]:
```suggestion
    "price_postapoc": "6000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 22]
```diff
@@ -163,6 +178,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_dexterity_1",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:181]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 23]
```diff
@@ -171,6 +187,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_dexterity_2",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:190]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 24]
```diff
@@ -179,6 +196,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_dexterity_3",
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:199]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 25]
```diff
@@ -187,6 +205,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_dexterity_4",
+    "price_postapoc": 600000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:208]:
```suggestion
    "price_postapoc": "6000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 26]
```diff
@@ -195,6 +214,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_intelligence_1",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:217]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 27]
```diff
@@ -203,6 +223,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_intelligence_2",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:226]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 28]
```diff
@@ -211,6 +232,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_intelligence_3",
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:235]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 29]
```diff
@@ -219,6 +241,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_intelligence_4",
+    "price_postapoc": 600000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:244]:
```suggestion
    "price_postapoc": "6000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 30]
```diff
@@ -227,6 +250,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_perception_1",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:253]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 31]
```diff
@@ -235,6 +259,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_perception_2",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:262]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 32]
```diff
@@ -243,6 +268,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_perception_3",
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:271]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 33]
```diff
@@ -251,6 +277,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_perception_4",
+    "price_postapoc": 600000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:280]:
```suggestion
    "price_postapoc": "6000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 34]
```diff
@@ -259,6 +286,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_speed_3",
+    "price_postapoc": 170000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:289]:
```suggestion
    "price_postapoc": "1700 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 35]
```diff
@@ -267,6 +295,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_speed_5",
+    "price_postapoc": 250000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:298]:
```suggestion
    "price_postapoc": "2500 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 36]
```diff
@@ -275,6 +304,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_speed_7",
+    "price_postapoc": 500000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:307]:
```suggestion
    "price_postapoc": "5000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 37]
```diff
@@ -283,6 +313,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_speed_10",
+    "price_postapoc": 800000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:316]:
```suggestion
    "price_postapoc": "8000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 38]
```diff
@@ -291,6 +322,7 @@
     "copy-from": "mring_copper",
     "type": "TOOL_ARMOR",
     "id": "mring_protection_2",
+    "price_postapoc": 100000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:325]:
```suggestion
    "price_postapoc": "1000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 39]
```diff
@@ -312,6 +344,7 @@
     "copy-from": "mring_silver",
     "type": "TOOL_ARMOR",
     "id": "mring_protection_4",
+    "price_postapoc": 200000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:347]:
```suggestion
    "price_postapoc": "2000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 40]
```diff
@@ -333,6 +366,7 @@
     "copy-from": "mring_gold",
     "type": "TOOL_ARMOR",
     "id": "mring_protection_6",
+    "price_postapoc": 400000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:369]:
```suggestion
    "price_postapoc": "4000 USD",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 41]
```diff
@@ -354,6 +388,7 @@
     "copy-from": "mring_platinum",
     "type": "TOOL_ARMOR",
     "id": "mring_protection_8",
+    "price_postapoc": 800000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_rings.json:391]:
```suggestion
    "price_postapoc": "8000 USD",
```

File: data/mods/Magiclysm/items/enchanted_wands.json [hunk 42]
```diff
@@ -5,7 +5,7 @@
     "type": "TOOL",
     "weight": "230 g",
     "volume": "500 ml",
-    "price": 10000,
+    "price": 40000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_wands.json:8]:
```suggestion
    "price": "400 USD",
```

File: data/mods/Magiclysm/items/enchanted_wands.json [hunk 43]
```diff
@@ -45,6 +45,7 @@
     "id": "wand_magic_missile_minor",
     "name": { "str": "minor wand of magic missile", "str_pl": "minor wands of magic missile" },
     "copy-from": "spell_wand",
+    "price_postapoc": 15000,
```
Comment (Maleclypse) [inline data/mods/Magiclysm/items/enchanted_wands.json:48]:
```suggestion
    "price_postapoc": "150 USD",
```

**Pull request merged**